### ISSUE TITLE
Anybody willing to maintain a fork?

### ISSUE THREAD

**Issue by chmac** [samuelmeuli/tmignore#19]
It looks like @samuelmeuli hasn't been active on this repo for over a year. If somebody wants to maintain a fork that would be awesome, it seems like a great piece of code. @Frizlab maybe? :-)

**Comment by Frizlab** [comment]
Okay I have started modernisation on https://github.com/Frizlab/tmignore/tree/develop, but I do _not_ guarantee I’ll be able to actually finish nor maintain this.

**Comment by chmac** [comment]
@Frizlab Totally understandable, the work of maintenance can be huge. What about posting an update here when you feel like the code is ready for use? What about enabling issues on your fork repo?

For anyone who's impatient and finds this issue, my research suggests that [heptapod](https://github.com/tg44/heptapod) is the best bet for something that works today, supports exclusions, and seems to be really simple (which I hope means less maintenance hassles and it Just Works™ for longer!).

**Comment by Frizlab** [comment]
Yes of course, I'll keep you posted. My goal is first to move all the old deps to new ones (e.g. use Apple's Argument Parser, a better logger, a better exec module, etc.)
Then I'll see what I can do in terms of evolution of the product itself.

**Comment by tomitrescak** [comment]
Can you add support for iCloud ignoring as well ... pretty please! ;)

**Comment by Frizlab** [comment]
I’d love to but if I remember correctly it’s not possible. Only files/folders whose name end with `.nosync` are ignored by iCloud AFAIK.
If I’m wrong, I’d be happy to add whatever code is necessary to have iCloud ignoring though.

**Comment by tomitrescak** [comment]
Yah, I bumped into this :/ no worries safe to ignore

**Comment by LudwigWS** [comment]
I'm curious about what happened to him, he hasn't had Github activity for over a year

**Comment by Reeywhaar** [comment]
I'm working on my own version of same functionality. You can check it in [reeywhaar/tmbliss](https://github.com/reeywhaar/tmbliss)

**Comment by reilem** [comment]
> I'm working on my own version of same functionality. You can check it in [reeywhaar/tmbliss](https://github.com/reeywhaar/tmbliss)

This works very well! I just used it to exclude all the gitignores files from my coding repos. Verified with `tmutil` and `mdfind` that it added everything it should have. Thanks!

**Comment by IohannRabeson** [comment]
Hi FYI I made tmignore-rs, a successor to tmignore. It's a direct replacement, it implements the same commands run, reset and list but also provides a new command monitor (and more) so you can start it and forget. It is also available on Homebrew.
I'm not using `tmutil` so my implementation is much faster - on a M4 pro with ~1M entries in the filesystem, tmignore with `tmutil` takes 14 minutes to exclude 667 paths, my implementation takes 2.5 seconds to do the exact same).
The repository is here: https://github.com/IohannRabeson/tmignore-rs
If you have suggestions do not hesitate to open an issue.

**Comment by chris-str-cst** [comment]
Since this thread is folks looking for a maintained alternative — I built one, so flagging it (disclosure: my tool). It's a **different approach**, not a drop-in fork, and the difference is kind of the point:

[**tmtidy**](https://github.com/chris-str-cst/tmtidy) excludes build dirs by **rule**, not by parsing `.gitignore`.

The reason I didn't go the gitignore route: `.gitignore` lists plenty of things you actually *want* backed up — `.env`, local secrets, dev databases, `.envrc`, machine-specific config. Those are ignored by git precisely because they're local and not reproducible, which is exactly what you'd want Time Machine to keep. Excluding everything in `.gitignore` risks dropping your only copy of them.

So tmtidy instead matches known regenerable build output — `node_modules`, `target`, `.venv`, `build`, `DerivedData`, Gradle/Maven caches, etc. — via baked-in rules across your code roots. A config with just `roots:` works out of the box.

Other differences from tmignore:
- **No daemon, no sudo** — writes the sticky `com_apple_backup_excludeItem` xattr directly (same one `tmutil addexclusion` sets).
- Optional **launchd agent** runs `scan` hourly as your user, so fresh clones get excluded before the next backup.
- Rust, single binary, Homebrew tap. macOS only (Apple Silicon + Intel).

Won't fit if you specifically want gitignore-driven exclusion — but if the goal is "stop Time Machine hoarding build output without risking my local-only files," that's the trade-off it makes on purpose. Wrote up the why here: https://gist.github.com/chris-str-cst/da5e8829f741b55a425968b50812b439

**Comment by IohannRabeson** [comment]
@chris-str-cst I did not looked how you really do it, but simply writing the com_apple_backup_excludeItem xattr is not enough to exclude a file from a Time Machine backup, even if `tmutil isexcluded` tells you it is.

**Comment by reeywhaar** [comment]
@IohannRabeson why is it so? Quick googling confirms xattr is fine enough. What are the pitfalls?

Problem with `tmutil addexclusion` is that it works slow for some reason. Like 1 second per file for some cases.

**Comment by IohannRabeson** [comment]
@reeywhaar Maybe it would be useful to actually check what you find on the internet no? You just assumed google was right? You never actually tried to verify?

**Comment by reeywhaar** [comment]
@IohannRabeson maybe less aggressive? I did a quick search, checked couple of links, exact time not to dig into some stranger words.

**Comment by IohannRabeson** [comment]
@reeywhaar Sorry I should have sticked more emojii 😃 I guess because I'm not mad at you 🙃
But I'm certainly disappointed 😞 to see you did not bothered to test and blindly accepted what a "quick googling" gave you 🔍✅😇
I bothered to check a little bit your project 🕵️‍♂️ and it seems you generated the whole code in just one day 2 monthes ago 🎉⚡️..
So you did not bother to write it ✍️❌, you did not bother to test it 🧪❌, but you bother to promote it 📣💯👏
Maybe I should have just let you promote your stuff knowing it probably don't even work? 🚀💩
I think I helped you enough I will stop this discussion with you now, so you will not be aggressed anymore 🙏🕊️😌

**Comment by chmac** [comment]
@IohannRabeson could you share what is necessary to exclude files? That way folks who land here and are looking for alternatives can know what to look out for when evaluating their choices.

**Comment by IohannRabeson** [comment]
@chmac What I found is tmutil is calling `CSBackupSetItemExcluded` to exclude a file. This function is opaque but it indeed sets the attribute mentionned previously but i does other mysterious things I ignore. 
I confirmed using `CSBackupSetItemExcluded` is working properly by doing backups test.
The same way I noticed setting the attribute was not enough: by doing backups and noticing the files supposedly excluded were not.
There is more about how I found `CSBackupSetItemExcluded` in this reddit post, I probably risk to be slapped by apple someday but well it's there now: https://www.reddit.com/r/rust/comments/1sitopa/i_rewrote_tmignore_in_rust_667_paths_in_25s/

**Comment by chmac** [comment]
@IohannRabeson Interesting. Thanks for the info. I'm using heptapod to manage my exclude paths, but I never actually thought to check if it's working or not! I will investigate a little. Thanks for the tip.

**Comment by IohannRabeson** [comment]
@chmac You can check using the Finder, you can explore the disk where the backups are stored and you can navigate in them like regular folders. I see heptapod seems to use rules as well, this is something I can probably implement for tmignore-rs. Also I checked how heptapod is excluding and they choose the safest option: calling `tmutil addexclusion`, I don't know a safest method than this one, but it's the slowest.

**Comment by chmac** [comment]
@IohannRabeson Okay, thanks for the info, and especially for checking on heptapod.
